src/cli/eofwrap.py

```diff
@@ -0,0 +1,339 @@
+                    parent_beacon_block_root=header.parent_beacon_block_root,
+                    requests_root=header.requests_root,
+                )
+                assert not fixture_block.ommers
+                assert not fixture_block.withdrawals
+                assert not fixture_block.deposit_requests
+                assert not fixture_block.withdrawal_requests
+                assert not fixture_block.consolidation_requests
+
+                for fixture_tx in fixture_block.txs:
+                    fixture_tx_dump = fixture_tx.model_dump()
+                    fixture_tx_dump.pop("ty")
+                    fixture_tx_dump.pop("data")
+                    tx = Transaction(
+                        type=fixture_tx.ty,
+                        input=fixture_tx.data,
+                        **fixture_tx_dump,
+                    )
+                    block.txs.append(tx)
+            elif isinstance(fixture_block, InvalidFixtureBlock):
+                block = Block(
+                    rlp=fixture_block.rlp,
+                    exception=fixture_block.expect_exception,
+                )
+            else:
+                raise TypeError("not a FixtureBlock")
+
+            test.blocks.append(block)
+
+        result = test.generate(
+            request=None,  # type: ignore
+            t8n=t8n,
+            fork=CancunEIP7692,
+            fixture_format=BlockchainFixture,
+        )
+        if traces:
+            print_traces(t8n.get_traces())
+        return result
+
+    def _validate_eof(self, container: Container, metrics: bool = True) -> bool:
+        eof_parse = EOFParse()
+
+        result = eof_parse.run(input=to_hex(container))
+        actual_message = result.stdout.strip()
+        if "OK" not in actual_message:
+            if metrics:
+                _inc_counter(self.metrics[self.VALIDATION_ERRORS], actual_message)
+            return False
+
+        return True
+
+
+# `no_type_check` required because OpcodeWithOperand.opcode can be `None` when formatting as a
+# string, but here it can never be `None`.
+@no_type_check
+def wrap_code(account_code: Bytes) -> Container:
+    """
+    Wraps `account_code` into a simplest EOF container, applying some simple heuristics in
+    order to obtain a valid code section termination.
+    """
+    assert len(account_code) > 0
+
+    opcodes = process_evm_bytes(account_code, allow_unknown=True)
+
+    if not opcodes[-1].opcode.terminating:
+        opcodes.append(OpcodeWithOperands(opcode=Op.STOP))
+
+    while len(opcodes) > 1 and opcodes[-2].opcode.terminating and opcodes[-1].opcode.terminating:
+        opcodes.pop()
+
+    bytecode = Bytecode()
+
+    for opcode in opcodes:
+        if opcode.operands:
+            # opcode.opcode[*opcode.operands] crashes `black` formatter and doesn't work.
+            bytecode += opcode.opcode.__getitem__(*opcode.operands)
+        else:
+            bytecode += opcode.opcode
```

Comment:
```suggestion
        bytecode += opcode.bytecode()
```
We could add this as a method to `OpcodeWithOperands`:
```python
@property
def bytecode(self) -> Bytecode:
    """Opcode as bytecode with its operands if any."""
    return self.opcode.__getitem__(*self.operands) if self.operands else self.opcode
```